### PR TITLE
Added library for atmosphere calculations

### PR DESCRIPTION
**Pull request by RomanBapst**
I need some of these function for the fixed wing performance work, but for the sake of not overloading I separated this from #22091

### CODE REVIEW

File: src/lib/atmosphere/atmosphere.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+ * @file atmosphere.cpp
+ *
+ */
+
+#include <geo/geo.h>
+#include "atmosphere.h"
+
+static constexpr float kTempRefKelvin = 15.f - CONSTANTS_ABSOLUTE_NULL_CELSIUS; // temperature at base height in Kelvin
+static constexpr float kTempGradient = -6.5f / 1000.f; // temperature gradient in degrees per meter
+static constexpr float kPressRefSeaLevelPa = 101325.f; // pressure at sea level in kPa
```
Comment (RomanBapst) [inline src/lib/atmosphere/atmosphere.cpp:44]: fix comment

File: src/lib/atmosphere/atmosphere.h [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+#ifndef PX4_SRC_LIB_ATMOSPHERE_ATMOSPHERE_H_
+#define PX4_SRC_LIB_ATMOSPHERE_ATMOSPHERE_H_
+
+/**
+* Calculate air density given air pressure and temperature.
+*
+* @param pressure_pa ambient pressure in Pa
+* @param temperature_celsius ambient temperature in degrees Celsius
+*/
+float getDensityFromPressureAndTemp(const float pressure_pa, const float temperature_celsius);
```
Comment (dagar) [inline src/lib/atmosphere/atmosphere.h:48]: For usage it might be slightly nicer in an `atmosphere` namespace?

atmosphere::getPressureFromAltitude(), etc

Comment (RomanBapst) [reply src/lib/atmosphere/atmosphere.h:51]: @dagar done

**Comment by RomanBapst** [comment]
@dagar I would keep them separate as atmosphere is very generic and applies to all vehicles. Airspeed is rather specific to fixed wings and pitot tubes.

**Comment by hendjoshsr71** [comment]
You should note somewhere you've only modeled the first layer of the US Standard Atmosphere 1976. 
AKA it only applies to 11,000m (which isn't that high up).

I have a branch for ArduPilot that adds the entire model that I never got around to getting into master :( .
Feel free to use as much of it that you want to [here](https://github.com/hendjoshsr71/ardupilot/pull/5/files#diff-15602edf26801ca0d26bdf6381a3154956bc950dd377a53d710231ccf42c6079R21)

**Comment by RomanBapst** [comment]
@hendjoshsr71 Thanks for the input, appreciate it! Why did your changes never make it into Ardupilot? Was it flash related?

**Comment by hendjoshsr71** [comment]
> @hendjoshsr71 Thanks for the input, appreciate it! Why did your changes never make it into Ardupilot? Was it flash related?

My version saved 1.5KB vs an inaccurate table on a PR that came up. That original PR was used in some NOAA research for balloons and think glider drops from 70-80k feet

It was dependent on another PR that never got in and I let it slide into the ether of bit rotting....

**Comment by RomanBapst** [comment]
@dagar Addressed your comments. I also flew this PR on two VTOLs until now and the data looks good.

**Review by dagar — APPROVED** [review]

**Pull request merged**